### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by quesurifn**
*Please check off the stories that have been completed in your homework submission. Feel free to add any additional notes to the PR description.*

### User stories to implement
 - [X] A user should not be able to complete a task when creating it.
 - [X] A user can filter their view by complete or incomplete tasks.
 - [X] A user's default view is filtered to show only incomplete tasks.
 - [X] A user should be able to keep multiple lists.
 - [X] A user should have access to only their tasks. (may require devise or a creative, lightweight auth system)
 - [X] A user viewing a task should see a history of all its changes. (task audit trail)
 - [X] A user should be able to see a history of all completed tasks and associated actions. (app audit trail)

### Cleanup tasks
- [X] Consolidate the repeated code between the create and edit feature specs for tasks.

### Review Status
**Note to candidate:** Do not edit this section

- [ ] Specs Passing
- [ ] Two approving reviews

## Note to reviewer: 
Normally I would add more tests including request, controller, model and more FE testing however, due to interviewing at 5+ companies this week and similar challenges I was unable to (time constraint).

### CODE REVIEW

**Comment by quesurifn** [comment]
@jamesconant My feedback for you: You should actually review the code before providing feedback. It is incredibly rude to ask someone to build to this and not even review. Also it's evident that in the review of the weather app, the person conducting the review didn't read the code. He asked a question that wouldn't have been asked had he read it. 

How would you overcome duplicate names in the weather api? I did... I used zip codes instead.

And asking me why I didn't use state management in an application with one input field? 

Wow. 

What an incredibly disrespectful waste of my time, my families time, and the time I spend freelancing on projects outside of my W2.

**Comment by jamesconant** [comment]
@quesurifn apologies for the delay in getting back to you. We take any and all feedback into serious consideration as we iterate on our interview processes here at Contently, so thank you for reaching out and providing your thoughts. 

Our prior policy on review comments was that we left them only when critical and likely to move the candidate forward after review. This was mainly due to the fact that we were drowning in submissions and have a lot of great candidates to review. We are fortunate to have that problem, but we also recognize that it might lead to unfortunate situations like this. As such, we have updated our policy going forward to leave feedback on every submission, regardless of candidate viability.

I can assure you that numerous engineers on our team reviewed your submissions, despite not all them leaving comments - as we do with all the engineers that express interest and submit assessments. Unfortunately, we are not able to proceed with every candidate and have to make our best judgment on when we should and should not.

In summary, however, I appreciate the feedback that we can and should be more communicative and we are striving to improve as much as we can along the way. Good luck in your continued job hunt and adventures!

Best,
James
_Head of Engineering_
